sunpy/data/_sample.py

```diff
@@ -78,50 +104,23 @@ def download_sample_data(overwrite=False):
         # Creating the directory for sample files to be downloaded
         sampledata_dir = Path(get_and_create_sample_dir())
 
-    dl = Downloader(overwrite=overwrite)
-
-    first_url = _base_urls[0]
-
     already_downloaded = []
-    for file_name in _sample_files.keys():
-        url = urljoin(first_url, file_name)
-        fname = sampledata_dir/file_name
-        # We have to avoid calling download if we already have all the files.
+    to_download = []
+    for url_file_name in _sample_files.keys():
+        fname = sampledata_dir/url_file_name
+        # We want to avoid calling download if we already have all the files.
         if fname.exists() and not overwrite:
             already_downloaded.append(fname)
         else:
-            dl.enqueue_file(url, filename=fname)
-
-    if dl.queued_downloads:
-        results = dl.download()
+            # URL and Filename pairs
+            to_download.append((url_file_name, fname))
+    if to_download:
+        results = _download_sample_data(_base_urls[0], to_download, overwrite=overwrite)
     else:
         return already_downloaded
 
-    if not results.errors:
-        return results
-    else:
-        log.info('Failed to download one or more sample data files, retrying with a mirror.')
-
-    for retry_url in _base_urls[1:]:
-        for i, err in enumerate(results.errors):
-            file_name = err.filepath_partial().name
-            log.debug(
-                f"Failed to download {_sample_files[file_name]} from {err.url}: {err.exception}")
-            # Overwrite the parfive error to change the url to a mirror
-            new_url = urljoin(retry_url, file_name)
-            results._errors[i] = _error(err.filepath_partial,
-                                        new_url,
-                                        err.exception)
+    # Something went wrong.
+    if results.errors:
+        results = _retry_sample_data(results)
 
-        results = dl.retry(results)
-
-        if not results.errors:
-            return results
-
-    for err in results.errors:
-        file_name = err.filepath_partial().name
-        log.debug(f"Failed to download {_sample_files[file_name]} from {err.url}: {err.exception}")
-        log.error(
-            f"Failed to download {_sample_files[file_name]} from all mirrors, the file will not be available.")
-
-    return results
+    return sorted(results + already_downloaded)
```

Comment:
Not sure if this is correct at all.

Reply:
iirc, I don't think it matters what order this comes back in, but I would need to check

Reply:
I meant more if I can a results to a already_downloaded.